src/ucs/datastruct/array.h

```diff
@@ -314,14 +308,14 @@ ucs_array_old_buffer_set_null(void **old_buffer_p)
  * @return Current capacity of the array
  */
 #define ucs_array_capacity(_array) \
-    ((_array)->capacity & UCS_ARRAY_CAP_MASK)
+    ((_array)->capacity)
 
 
 /**
  * @return Whether this is a fixed-length array
  */
 #define ucs_array_is_fixed(_array) \
-    ((_array)->capacity & UCS_ARRAY_CAP_FLAG_FIXED)
+    ((_array)->is_fixed)
```

Comment:
maybe remove these wrappers now?

Reply:
- imo, using wrappers looks cleaner
- can be on the same line

Reply:
While working on the fix I've already considered both ideas: remove macros and put it on the same line, and finally I decided to keep as it is now, because it's consistent with the rest of the macros in this file:
```
#define ucs_array_length(_array) \
    ((_array)->length)

#define ucs_array_elem(_array, _index) \
    ((_array)->buffer[_index])

#define ucs_array_begin(_array) \
    ((_array)->buffer)
```
So in order to keep this file consistent, I would just keep it as it is now.
Unless you have strong opinion on that

Reply:
ok